core/src/main/java/org/mule/expression/DefaultExpressionManager.java

```diff
@@ -656,7 +659,7 @@ public Object match(String token)
         else if (!match.get())
         {
             throw new InvalidExpressionException(expression,
-                "Expression string is not an expression.  Use isExpression(String) to validate first");
+                "Expression '"+expression+"' is not an expression.  Use isExpression(String) to validate first");
```

Comment:
I don't think the "string" part should be removed, because otherwise it sounds like the expression is not an expression which is confusing.

Reply:
(y)